### ISSUE TITLE
Sprite is black

### ISSUE THREAD

**Issue by Farnoway** [viniciusgerevini/godot-aseprite-wizard#40]
Hello! 
I have been using aseprite wizard for a while now, but I have never seen anything like this. 
When I run my game on Godot, the player looks great, as it should be:
![image](https://user-images.githubusercontent.com/95051206/143495001-0cdbe60b-8e51-410f-810d-6fbb8fc9b922.png)
But when I export the game, it looks like this: 
![image](https://user-images.githubusercontent.com/95051206/143495171-6bfd9081-41a7-4605-b48e-85270e117816.png)
Im still trying to see what I did wrong this time with the addon, because this is the first time that this has happened to me.

**Comment by viniciusgerevini** [comment]
Hello @Farabry . I see you closed the issue. Did you find what was the problem?

**Comment by markman4897** [comment]
@viniciusgerevini I got the same problem with one specific project. If I start a new project and add Aseprite Wizard it works fine, but while trying to port over files from the project with the problem to the new project I just can't seem to figure out when exactly this problem starts occurring.
I can describe this here or open another issue and add the project files too.

**Comment by tavurth** [comment]
Also ran into this just now, can we re-open this at least until we have a fix?

```
Godot Engine v3.4.1.stable.official.aa1b95889 - https://godotengine.org
OpenGL ES 3.0 Renderer: Intel(R) Iris(TM) Plus Graphics 655
OpenGL ES Batching: ON

ERROR: Condition "!f" is true. Returned: ERR_CANT_OPEN
   at: _load_data (scene/resources/texture.cpp:466)
```

**Comment by tavurth** [comment]
While exporting via command line I see the following errors:

```
WARNING: Loaded resource as image file, this will not work on export: 'res://icon.png'. Instead, import the image file as an Image resource and load it normally as a resource.
     at: load (core/image.cpp:2051)
ERROR: Condition "_first != nullptr" is true.
   at: ~List (./core/self_list.h:108)
ERROR: Condition "_first != nullptr" is true.
   at: ~List (./core/self_list.h:108)
```

**Comment by tavurth** [comment]
Interestingly when I culled out the `addons` directory from my export the project is now working, but only when run from the command line. When I run the app directly (OSX) the sprite is still black.

**Comment by tavurth** [comment]
Ok, what fixed it for me was choosing the following style export settings:

<img width="460" alt="Screenshot 2021-12-20 at 22 09 03" src="https://user-images.githubusercontent.com/100964/146819678-e969f08d-012d-4951-a033-bb55610af786.png">

Perhaps for some reason the `.res` file or the `.png` file are not getting required properly by the `AnimatedSprite`?

**Comment by tavurth** [comment]
Can we open this issue again? @viniciusgerevini

**Comment by viniciusgerevini** [comment]
Hello @tavurth . Thanks for reaching out. As you are dealing with `.res` files I guess you are using the wizard screen to import your animations, right?

The wizard screen embeds the png as part of the resource, so the external png file shouldn't even be required for it to work.

I tried to simulate this issue a few times, but I couldn't make it (classic works on my machine). Would you be able to provide a simple project where this issue happens?

**Comment by tavurth** [comment]
@viniciusgerevini Yes sure I can provide my current sources for a Christmassy game I'm working on, I'll send them over to your email address rather than posting here directly.

**Comment by viniciusgerevini** [comment]
Thanks @tavurth . As I mentioned in the email, I've done a few tests and they all worked .
1 - Exported a MacOS build from a Linux machine (already setup).
2 - Exported a Linux build from a Linux machine (already setup).
3 - Exported a MacOS build from a MacOS machine. (had warnings on Godot startup due to Aseprite not being configured, worked in a next clean run)
4 - Exported using your script via CLI from a MacOS machine. I've got the same warnings you did.

Some details:
- Linux machine: Ubuntu 21.10, Godot 3.4.2, Aseprite 1.2.30
- Macbook: macOS Catalina 10.15.7. Godot 3.4.2, Aseprite 1.2.9

I can see you were using Godot 3.4.1 before, but I guess you may have updated it in the meantime. 

The importer does have a workaround for inserting the image, which could justify the black box. However I wasn't able to make it happen on my machines. 

@markman4897 are you still having this issue?

**Comment by markman4897** [comment]
@viniciusgerevini since I couldn't wait for the fix this long and I didn't have time to rewrite the project I used another Aseprite importer in the project. However, I saved two projects in their broken state just in case someone would want to take a look at them. Want me to send the files to you over email?

**Comment by viniciusgerevini** [comment]
@markman4897 Sure. Thanks!

**Comment by markman4897** [comment]
@viniciusgerevini I'm working on Linux (Pop!_OS 21.04 64bit) while the version of Godot was mixed, it was an old project that I started with Godot 3.2.x I think and then I always used the latest version, same thing with Aseprite.
I was testing exports to Linux and HTML5 platforms.

The funny thing was that in editor everything looked just fine. The project I was working on was about a year old and everything worked great. Then I picked up where I left off recently and updated the plugin and everything I imported with the new plugin was rendered as black boxes in the exported version (while showing up just fine in the editor).
Then I tried to import new assets in a new empty project with a fresh install of Aseprite Wizard and that worked fine, then I moved those imported files to the old project and it didn't work anymore in the exported version.
Then I tried exporting with an old version of Aseprite Wizard in another project (which also worked fine) and moving those files to the project that had problems with the updated plugin, and that worked out fine.
I'm sorry this might be written in a confusing way, but the cases when it started to magically work were pretty weird.

Anyways, below are some screenshots of the problematic projects (ignore the goofy art, it's a silly easter egg)

Project 1 editor
![image](https://user-images.githubusercontent.com/8660220/147398444-3165bfb6-4be3-4c2c-8610-4a0a78282021.png)
Project 1 linux export
![image](https://user-images.githubusercontent.com/8660220/147398456-ee7243ed-62d3-476c-bd03-7d1c3a5740d0.png)
Project 2 editor
![image](https://user-images.githubusercontent.com/8660220/147398429-3ac2e3eb-1299-4919-b286-04c5b915fd8b.png)
Project 2 linux export
![image](https://user-images.githubusercontent.com/8660220/147398434-a2e229da-d1ee-4f6e-9615-c0f1d76fe217.png)

**Comment by Farabry** [comment]
Ohhhh, sorry I didn't saw this earlier. Gonna explain what I did to fix this:

-I was moving the folder where the _.res_ and _.png_ files were to another location, so it was no more the _Output folder_ location. Just make sure that you put the folder or files at the correct path in your res://. Im pretty sure that will make it. 

If you have a question regarding this, im gonna be here.

**Comment by viniciusgerevini** [comment]
Thanks all. Unfortunately all my VMs and machines seem to not cause the issue. I noticed that the importer does not include the image in the list of generated resources and that could potentially cause some issues, but I can't be sure that is what's causing this one. I'll include that fix a future version.

I'll leave this issue open. If any one comes across this issue again I would ask you to send me the .import folder and the broken binary, so I can run some extra tests

**Comment by terijaki** [comment]
![Screenshot 2022-02-28 at 09 57 58](https://user-images.githubusercontent.com/590522/155953883-fb1aa14d-1fbc-44be-be06-0510c261abd1.png)
I had the same issue described here. Unchecking "**remove source files**" in the config solved it for me!

My issue:
- In the editor everything was fine
- Exporting the project results in black boxes

Note, when I delete the ".import" folder of my project, the animations are all black in the editor as well, until I reimport them. But afterwards the export is still black.

So for now I am avoiding "removing source files".

**Comment by viniciusgerevini** [comment]
Thanks @terijaki . I noticed this issue last week with files imported via the docks. I realised the SpriteFrames is actually using the .stex file from the import forlder, so we do need to keep the png file at least. I'll be releasing a fix where only the json will be removed.

This should impact only resources imported manually. I believe the issue with the automatic importer is slightly different. Which method are you using in your project?
